### DISCUSSION TITLE
select don't selected options use set value by function not by user

### DISCUSSION THREAD

**Discussion by salahhusa9** [livewire/livewire#8026] [4. General]
I create a select `<select wire:model.live="selectedLiveBouquets">` , i have a btn with wire:click for selectAll options (it's multiple select) by call selectAll and set values in $selectedLiveBouquets, i print in view i see values of $selectedLiveBouquets but he don't selected (it's work in livewire 2)

**Comment by devhau** [comment]
me too,
i same this issue

**Comment by danie-ramdhani** [comment]
can you provide you code?

maybe look like this??

```php
public array $selectedLiveBouquets = [];

public function selectChoice(): array
{
    return [
        'foo' => 'Foo',
        'bar' => 'Bar',
    ];
}

public function selectAll(): void
{
    $this->selectedLiveBouquets = array_keys($this->selectChoice());
}
```

view:
    
```blade
<select wire:model.live="selectedLiveBouquets" multiple>
    @foreach ($this->selectChoice() as $key => $value)
        <option value="{{ $key }}">{{ $value }}</option>
    @endforeach
</select>

<button wire:click="selectAll">
    Select All
</button>

<span>{{ json_encode($selectedLiveBouquets) }}</span>
```

> [!NOTE]
> livewire is using discussion tab to track ideas/bug. if you need help just go to livewire discord / discussion tab (see https://github.com/livewire/livewire/discussions/3147) and please close your issue.

**Reply by salahhusa9** [reply]
yes it's like this, this code it's not work, he don't show options as selected

**Reply by salahhusa9** [reply]
```html

  <select wire:model.live="selectedLiveBouquets" size="10" multiple 
  >
      @foreach ($this->selectChoice() as $key => $value)
          <option value="{{ $key }}">{{ $value }}</option>
      @endforeach
  </select>
  
  <button wire:click="selectAll">
      Select All
  </button>
  
  <span>{{ json_encode($selectedLiveBouquets) }}</span>

```

**Reply by danie-ramdhani** [reply]
works for me:

https://wirebox.app/b/xegnk

**Reply by danie-ramdhani** [reply]
maybe (just maybe) you forget that livewire need a single root element (https://livewire.laravel.com/docs/quickstart#writing-the-view)

**Comment by PhiloNL** [comment]
Hi all,
I just tried to re-create the example you've provided and this seems to work:
https://wirebox.app/b/xgrl8

**Reply by salahhusa9** [reply]
yes, it's work in online but i use this code in local and he don't work hhhhh

**Reply by salahhusa9** [reply]
it's work now, the problem is i use Alpine cdn

**Reply by salahhusa9** [reply]
my case it's complicated than that. but it's working with v2, i have many selects (3 parent and each one has Childrens when i choise parent Childrens show in Children select and selected automatically or by click selectAll, now if i click selectAll it's work but if i click parent (for automaticaly select children) don't work in code i set options and call selectAll

**Comment by salahhusa9** [comment]
![Packages-Form-ReborNott-Dashboard](https://github.com/livewire/livewire/assets/64494826/e51ab7da-0774-4421-ad82-651744169a21)

**Comment by lotje-kinable** [comment]
I created a form and tried to update a value by using $this->form->value = 5; (an id) and it did not work. All others do work: inputs, checkboxex, etc. But not the select...

After searching for hours, I decided to try the 'key'  as shown here. [Dependent Select Dropdowns](https://livewire.laravel.com/docs/wire-model#dependent-select-dropdowns) and IT WORKS!

**Comment by kriit24** [comment]
it does not work also with key
it should work like just like the html + php -  that he sends the first option key
